modules/phase_field/src/materials/GammaAndLAnisoGauss.C

```diff
@@ -0,0 +1,569 @@
+                         (1.0 * 2.0 * vsmalbay))) /
+                           normvsmallbavalue +
+                       (((-1.0 * miubalibxR * vsmalbax) *
+                         ((1.0 / (2.0 * normvsmallba)) * (2.0 * 1.0 * vsmalbay))) /
+                        normvsmallbavalue) +
+                       (((-1.0 * miubalibzR * vsmalbaz) *
+                         ((1.0 / (2.0 * normvsmallba)) * (2.0 * 1.0 * vsmalbay))) /
+                        normvsmallbavalue));
+                  const ADReal finalGaussian3plus = _ADDGaussian ? Gaussian3plus : -Gaussian3plus;
+                  const ADReal Gaussian4plus =
+                      _amplitudeScale * amplitudes * S_switch_value * _sharpness *
+                      std::exp(exponent) *
+                      (((1.0 * miubalibzR * normvsmallba) -
+                        (miubalibzR * vsmalbaz * (1.0 / (2.0 * normvsmallba)) *
+                         (1.0 * 2.0 * vsmalbaz))) /
+                           normvsmallbavalue +
+                       (((-1.0 * miubalibxR * vsmalbax) *
+                         ((1.0 / (2.0 * normvsmallba)) * (2.0 * 1.0 * vsmalbax))) /
+                        normvsmallbavalue) +
+                       (((-1.0 * miubalibyR * vsmalbay) *
+                         ((1.0 / (2.0 * normvsmallba)) * (2.0 * 1.0 * vsmalbax))) /
+                        normvsmallbavalue));
+                  const ADReal finalGaussian4plus = _ADDGaussian ? Gaussian4plus : -Gaussian4plus;
+
+                  aniso += finalGaussian;
+                  anisox += finalGaussian2;
+                  anisoy += finalGaussian3;
+                  anisoz += finalGaussian4;
+                  anisoL += finalGaussianL;
+                  anisoxplus += finalGaussian2plus;
+                  anisoyplus += finalGaussian3plus;
+                  anisozplus += finalGaussian4plus;
+                }
+              }
+              sumGgamma += aniso * Val;
+              sumGgammax += anisox * Val;
+              sumGgammay += anisoy * Val;
+              sumGgammaz += anisoz * Val;
+              sumGgammaxplus += anisoxplus * Val;
+              sumGgammayplus += anisoyplus * Val;
+              sumGgammazplus += anisozplus * Val;
+              sumL += anisoL * Val;
+            }
+
+  _L[_qp] = std::fabs(sumL / sumval);
+  _L_EN[_qp] = MetaPhysicL::raw_value(sumL / sumval);
+  _Ggamma[_qp] = std::fabs(sumGgamma / sumval);
+  _dGgamma_minus[_qp] = {sumGgammax / sumval, sumGgammay / sumval, sumGgammaz / sumval};
+  _dGgamma_plus[_qp] = {sumGgammaxplus / sumval, sumGgammayplus / sumval, sumGgammazplus / sumval};
+
+  if (sumval == 0.0)
+  {
+    _L[_qp] = _L_BASE;
+    _L_EN[_qp] = MetaPhysicL::raw_value(_L_BASE * 1.0);
+    _Ggamma[_qp] = _GgammaBASE;
+    _dGgamma_minus[_qp] = {0.0, 0.0, 0.0};
+    _dGgamma_plus[_qp] = {0.0, 0.0, 0.0};
+  }
+  else if (_Ggamma[_qp] == 0.0)
+  {
+    _L[_qp] = _L_BASE;
+    _L_EN[_qp] = MetaPhysicL::raw_value(_L_BASE * 1.0);
```

Comment:
where are these non AD material properties being used? We generally never need to do that

Reply:
I was using TotalFreeEnergy auxkernel that required non-AD materials. But, I made a new AD version ADTotalFreeEnergy; so I do not have anymore non-AD materials